sdks/python/apache_beam/runners/common.py

```diff
@@ -558,16 +568,33 @@ def __init__(self, window_fn, main_receivers, tagged_receivers):
     self.main_receivers = main_receivers
     self.tagged_receivers = tagged_receivers
 
+    # TODO: remove if block after per-element-output-counter released
+    # TODO: [BEAM-3937]
+    experiments = RuntimeValueProvider.get_value('experiments', str, [])
+    if 'outputs_per_element_counter' in experiments:
+      self.has_per_element_output_count = True
+      per_element_output_counter_name = \
+        CounterName('per-element-output-count-meta', step_name=operation_name)
+      self.per_element_output_counter = counter_factory.get_counter(
+          per_element_output_counter_name, Counter.DISTRIBUTION_METADATA)
+    else:
+      self.has_per_element_output_count = False
+
   def process_outputs(self, windowed_input_element, results):
     """Dispatch the result of process computation to the appropriate receivers.
 
     A value wrapped in a TaggedOutput object will be unwrapped and
     then dispatched to the appropriate indexed output.
     """
     if results is None:
+      # TODO: remove if block after per-element-output-counter released
+      if self.has_per_element_output_count:
+        self.per_element_output_counter.update(0)
       return
 
+    output_element_output = 0
     for result in results:
+      output_element_output += 1
```

Comment:
Why not just do `output_element_output = len(results)`?  Also, maybe rename to `output_element_count`?

Reply:
ah yes, I'll modify it in next revision. Thanks!